docs/language/values-and-types.mdx

```diff
@@ -405,14 +405,26 @@ let aNumber = 0x436164656E636521
 // `aNumber` has type `Int`
 ```
 
-Address can also be created using a byte array.
+Address can also be created using a byte array or string.
 
 ```cadence
 // Declare an address with hex representation as 0x436164656E636521.
 let someAddress: Address = Address.fromBytes([67, 97, 100, 101, 110, 99, 101, 33])
 
 // Invalid: Provided value is not compatible with type `Address`. The function panics.
 let invalidAddress: Address = Address.fromBytes([12, 34, 56, 11, 22, 33, 44, 55, 66, 77, 88, 99, 111])
+
+// Declare an address with the string representation as "0x436164656E636521".
+let addressFromString: Address = Address.fromString("0x436164656E636521")
+
+// Declare an address with the string representation without the "0x" prefix.
+let addressFromStringWithoutPrefix: Address = Address.fromString("436164656E636521")
```

Comment:
Hmm, I am not sure how I feel about this. This kind of "adaptive" API behavior feels very Javascript-y to me, for lack of a better way of phrasing it. In general I think it's better to have a simple, consistent behavior for these kinds of library functions, and not have them do anything other than exactly what they say on the tin. Since `toString` includes the `0x` prefix in the output, IMO `fromString` should require it, and if it's not present then treat the string as invalid input.

Reply:
Agreed, the function should only accept the format that was generated by `toString`, i.e. require the `0x` prefix to indicate the hex format. This reduces potential confusion and safety issues, where e.g. a developer could intend an input like "1234" to mean "0x1234", whereas another could understand it to be in decimal format.

Reply:
Done, now we return Nil for those cases.